src/main/java/net/snowflake/client/jdbc/RestRequest.java

```diff
@@ -204,13 +205,20 @@ public static CloseableHttpResponse execute(
         httpRequest.setURI(builder.build());
 
         response = httpClient.execute(httpRequest);
-      } catch (Exception ex) {
+      } catch (IllegalStateException ex) {
         // if exception is caused by illegal state, e.g shutdown of http client
-        // because of closing of connection, stop retrying
-        if (ex instanceof IllegalStateException) {
-          throw new SnowflakeSQLLoggedException(
-              null, ErrorCode.INVALID_STATE, ex, /* session = */ ex.getMessage());
-        }
+        // because of closing of connection, then fail immediately and stop retrying.
+        throw new SnowflakeSQLLoggedException(
+            null, ErrorCode.INVALID_STATE, ex, /* session = */ ex.getMessage());
+
+      } catch (SSLException ex) {
+        // if an SSL issue occurs like an SSLHandshakeException then fail
```

Comment:
Can we do for a specific SSL exception like SSLHandshakeException instead of generic SSLException?

Reply:
This is more specific than what it used to be (Exception). There are only four other exceptions that extend [SSLException](https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLException.html) including SSLHandshakeException. The others can possibly occur in different environments for different reasons. If you want me to explicitly state the exceptions, then I would just change it to this:
`catch (SSLHandshakeException | SSLKeyException | SSLPeerUnverifiedException | SSLProtocolException ex)`

What do you think?

Reply:
yes. Because title of the issue mention a specific exception. so better to spell out so no regression happen.

Reply:
Sounds good. Made the changes